libbeat/tests/system/base.py

```diff
@@ -17,3 +21,115 @@ def setUpClass(self):
             "winlogbeat"
         ]
         super(BaseTest, self).setUpClass()
+
+    def clean(self):
+
+        # Delete existing indices and aliases
+        try:
+            self.es.transport.perform_request('DELETE', "/*")
+        except:
+            pass
+
+        # Delete templates
+        try:
+            self.es.transport.perform_request('DELETE', "/_template/*")
+        except:
+            pass
+
+        # Delete any existing policy
+        policies = self.es.transport.perform_request('GET', "/_ilm/policy")
+        for policy, _ in policies.items():
+            try:
+                self.es.transport.perform_request('DELETE', "/_ilm/policy/" + policy)
+            except:
+                pass
```

Comment:
The clean methods are pretty heavy operations, basically removing everything.

Ultimately each test should have unique names for resources, such that tests can be run in isolation. This also helps with introspection on errors (assuming the test logs resource names being used). A test should clear it's resources only on success.

We could reduce impact be always prefixing resource names with `beatstest-`.

Resource handling and cleanup likely requires a many more changes across a multitute of tests. Maybe better to handle this in a separate PR.

Reply:
I agree that this refactoring could go a lot further, I agree even more that this would be out of scope.

The current behavior for the clean methods, only deletes resources for given names. This can easily lead to passing in a wrong name, not deleting the resource and then running a test that checks a resource is there. The current behavior therefore is error prone. 

Running tests in isolation also means, that every test is responsible for setting up its environment _before_ running, so cleaning out everything should not have any impact on other tests. 

However, to prevent unpredicted side effects, I'll update to pass a `prefix` and only delete resources with this prefix. 


> A test should clear its resources only on success.

I'd say a test should clear resources _before_ running, to ensure a clean setup.

Reply:
See https://github.com/elastic/beats/pull/12065#discussion_r282040566